### PR TITLE
HMAN 21: Database creation for Hearing Management Component

### PR DESCRIPTION
**Pull request by Abi-Laver**
https://tools.hmcts.net/jira/browse/HMAN-21

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/resources/application.yaml [hunk 1]
```diff
@@ -10,34 +10,36 @@ management:
...
-#            non_contextual_creation: true
+  config:
+    import: "optional:configtree:/mnt/secrets/"
+  jackson:
+    property-naming-strategy: SNAKE_CASE
+  datasource:
+    driver-class-name: org.postgresql.Driver
+    url: jdbc:postgresql://${HMC_HEARING_MANAGEMENT_DB_HOST:localhost}:${HMC_HEARING_MANAGEMENT_DB_PORT:5060}/${HMC_HEARING_MANAGEMENT_DB_NAME:hmc_hearing_service}${HMC_HEARING_MANAGEMENT_DB_OPTIONS:}
+    username: ${HMC_HEARING_MANAGEMENT_DB_USERNAME:ccd}
+    password: ${HMC_HEARING_MANAGEMENT_DB_PASSWORD:ccd}
```
Comment (danlysiak) [inline src/main/resources/application.yaml:24]: `hmc` would be a more appropriate default than `ccd`

File: src/main/resources/application.yaml [hunk 2]
```diff
@@ -10,34 +10,36 @@ management:
...
+      ddl-auto: none
+    properties:
+      hibernate:
+        jdbc:
+          time_zone: UTC
+      org:
+        hibernate:
+          envers:
+            audit_table_suffix: _audit
+            store_data_at_delete: true
```
Comment (danlysiak) [inline src/main/resources/application.yaml:39]:
```suggestion
```

File: src/main/resources/db/migration/V20211001_21__HMAN-21-HearingService.sql [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+                         auto_list_flag boolean not null,
+                         hearing_type varchar(40) not null,
+                         required_duration_in_minutes integer not null,
+                         hearing_priority_type varchar(60) not null,
+                         number_of_physical_attendees integer,
+                         hearing_in_welsh_flag boolean,
+                         private_hearing_required_flag boolean,
+                         lead_judge_contract_type varchar(70),
+                         first_date_time_of_hearing_must_be timestamp without time zone,
+                         hmcts_service_id varchar not null,
```
Comment (danlysiak) [inline src/main/resources/db/migration/V20211001_21__HMAN-21-HearingService.sql:11]:
```suggestion
                         hmcts_service_id varchar(4) not null,
```
Based on the pattern `^\w{4}$` we can set this to 4

**Review by danlysiak — COMMENTED** [review]
A few comments mostly about naming conventions

**Review by noronhaa — APPROVED** [review]

**Pull request merged**